modules/users/server/models/user.server.model.js

```diff
@@ -121,20 +146,126 @@ UserSchema.methods.hashPassword = function(password) {
 /**
  * Create instance method for authenticating user
  */
-UserSchema.methods.authenticate = function(password) {
+UserSchema.methods.authenticate = function (password) {
 	return this.password === this.hashPassword(password);
 };
 
 /**
+ * Create instance method for authenticating user
+ */
+UserSchema.methods.setIdentity = function (provider, id, accessToken, refreshToken, providerData) {
+	var found = false;
+	var auth = {
+		provider: provider,
+		id: id
+	};
+	if (!this.identities) {
+		this.identities = [];
+	}
+	this.identities.forEach(function (identity) {
+		if (identity.provider === provider && identity.id === id) {
+			auth = identity;
+			found = true;
+		}
+	});
+	auth.accessToken = accessToken || null;
+	auth.refreshToken = refreshToken || null;
+	auth.providerData = providerData || null;
+	if (!found) {
+		this.identities.push(auth);
+	}
+};
+
+UserSchema.methods.getIdentity = function (provider, id) {
+	var auth = null;
+	if (this.identities) {
+		this.identities.forEach(function (identity) {
+			if (identity.provider === provider && (identity.id === id || id === undefined || id === null )) {
+				auth = identity;
+			}
+		});
+	}
+	return auth;
+};
+
+UserSchema.methods.removeIdentity = function (provider, id) {
+	if (this.identities) {
+		var c = this.identities.length - 1;
```

Comment:
Simply it's safer when you remove elements, as you don't have to jump with incrementor.